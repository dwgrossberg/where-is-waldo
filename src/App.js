import { ThemeProvider } from "@emotion/react";
import React, { useState, useEffect } from "react";
import { Routes, Route, HashRouter } from "react-router-dom";
import { db } from "./firebase";
import { getDocs, collection, query, where } from "firebase/firestore";
import ColorTabs from "./components/ColorTabs";
import Footer from "./components/Footer";
import Puzzle from "./components/Puzzle";
import theme from "./theme";
import Home from "./pages/Home";
import BestTimes from "./pages/BestTimes";
import waldo1 from "./assets/waldo1.jpg";
import waldo2 from "./assets/waldo2.jpg";
import waldo3 from "./assets/waldo3.jpg";
import waldo4 from "./assets/waldo4.jpg";
import waldo5 from "./assets/waldo5.jpg";
import waldo6 from "./assets/waldo6.jpg";
import waldo from "./assets/waldo.jpg";
import wizard from "./assets/wizard.jpg";
import wenda from "./assets/wendy.jpg";
import odlaw from "./assets/odlaw.jpg";
import "./styles/App.scss";

const App = () => {
  const [puzzles, setPuzzles] = useState([
    {
      img: waldo1,
      id: "waldo1",
      level: 1,
      characters: [
        { name: "waldo", img: waldo },
        { name: "wizard", img: wizard },
        { name: "wenda", img: wenda },
        { name: "odlaw", img: odlaw },
      ],
    },
    {
      img: waldo2,
      id: "waldo2",
      level: 2,
      characters: [
        { name: "waldo", img: waldo },
        { name: "wizard", img: wizard },
        { name: "wenda", img: wenda },
        { name: "odlaw", img: odlaw },
      ],
    },
    {
      img: waldo3,
      id: "waldo3",
      level: 3,
      characters: [
        { name: "waldo", img: waldo },
        { name: "wizard", img: wizard },
        { name: "wenda", img: wenda },
        { name: "odlaw", img: odlaw },
      ],
    },
    {
      img: waldo4,
      id: "waldo4",
      level: 4,
      characters: [
        { name: "waldo", img: waldo },
        { name: "wizard", img: wizard },
        { name: "wenda", img: wenda },
        { name: "odlaw", img: odlaw },
      ],
    },
    {
      img: waldo5,
      id: "waldo5",
      level: 5,
      characters: [
        { name: "waldo", img: waldo },
        { name: "wizard", img: wizard },
        { name: "wenda", img: wenda },
        { name: "odlaw", img: odlaw },
      ],
    },
    {
      img: waldo6,
      id: "waldo6",
      level: 6,
      characters: [
        { name: "waldo", img: waldo },
        { name: "wizard", img: wizard },
        { name: "wenda", img: wenda },
        { name: "odlaw", img: odlaw },
      ],
    },
  ]);

  const [bestTimes, setBestTimes] = useState([]);

  useEffect(() => {
    const getTimes = async () => {
      const querySnapshot = await getDocs(collection(db, "puzzles"));
      querySnapshot.forEach((doc) => {
        console.log(doc.id, " => ", doc.data());
      });
    };
    getTimes();
  }, []);

  return (
    <div className="App">
      <ThemeProvider theme={theme}>
        <HashRouter>
          <ColorTabs />
          <Routes>
            <Route path="/" element={<Home puzzles={puzzles} />} />
            <Route path="/puzzles/:id" element={<Puzzle puzzles={puzzles} />} />
            <Route path="/best-times" element={<BestTimes />} />
          </Routes>
        </HashRouter>
        <Footer />
      </ThemeProvider>
    </div>
  );
};

export default App;
