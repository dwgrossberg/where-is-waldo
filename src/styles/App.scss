.App {
  min-height: (100vh);
  width: 100%;
  height: 500px;
  gap: 20px;
  display: grid;
  align-items: stretch;
  justify-items: center;
  grid-template: 75px 1fr 50px / 1fr;
  padding: 30px;
  position: relative;
}

.Footer {
  grid-area: 3 / 1 / 4 / 2;
  height: 100%;
  margin-top: 10px;
  z-index: 10;
}
