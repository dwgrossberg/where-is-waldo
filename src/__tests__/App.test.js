import { render, screen, cleanup } from "@testing-library/react";
import App from "../App";

describe("App component", () => {
  afterEach(cleanup);
  it("snapshot test", () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });

  it("renders correct heading", () => {
    render(<App />);
    expect(screen.getByRole("heading").textContent).toMatch(/Where's Waldo?/i);
  });

  it("displays the correct number of puzzles", () => {
    render(<App />);
    expect(screen.getAllByAltText(/waldo/i)).toHaveLength(6);
  });
});
