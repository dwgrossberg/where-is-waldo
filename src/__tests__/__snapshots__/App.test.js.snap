// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component snapshot test 1`] = `
<div>
  <div
    class="App"
  >
    <div
      class="MuiBox-root css-6ga7a9-ColorTabs"
    >
      <div
        class="MuiTabs-root css-1ujnqem-MuiTabs-root"
      >
        <div
          class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
          style="overflow: hidden; margin-bottom: 0px;"
        >
          <div
            aria-label="primary tabs example"
            class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
            role="tablist"
          >
            <a
              aria-current="page"
              aria-selected="true"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-k9gjoh-MuiButtonBase-root-MuiTab-root active"
              href="#/"
              role="tab"
              tabindex="0"
            >
              Home
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </a>
            <a
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-k9gjoh-MuiButtonBase-root-MuiTab-root"
              href="#/best-times"
              role="tab"
              tabindex="-1"
            >
              Best Times
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </a>
          </div>
          <span
            class="MuiTabs-indicator css-hioh9d-MuiTabs-indicator"
            style="left: 0px; width: 0px;"
          />
        </div>
      </div>
      <h1
        class="css-1rux908-ColorTabs"
      >
        Where's Waldo?
      </h1>
    </div>
    <div
      class="Home css-3ln04g-Home"
    >
      <p
        class="css-14o6217-Home"
      >
        Choose a puzzle to get started!
      </p>
      <a
        class="css-10dzrpy-Home"
        href="#/puzzles/waldo1"
      >
        <img
          alt="waldo1"
          class="puzzle-img css-e4yxh9-Home"
          loading="lazy"
          src="waldo1.jpg"
        />
        <div
          class="css-1b3wpv8-Home"
        >
          <p>
            Level 
            1
          </p>
          <div
            class="css-bshjkg-Home"
          >
            <img
              class="css-1cjt916-Home"
              src="waldo.jpg"
            />
            <img
              class="css-1cjt916-Home"
              src="wizard.jpg"
            />
            <img
              class="css-1cjt916-Home"
              src="wendy.jpg"
            />
            <img
              class="css-1cjt916-Home"
              src="odlaw.jpg"
            />
          </div>
        </div>
      </a>
      <a
        class="css-10dzrpy-Home"
        href="#/puzzles/waldo2"
      >
        <img
          alt="waldo2"
          class="puzzle-img css-e4yxh9-Home"
          loading="lazy"
          src="waldo2.jpg"
        />
        <div
          class="css-1b3wpv8-Home"
        >
          <p>
            Level 
            2
          </p>
          <div
            class="css-bshjkg-Home"
          >
            <img
              class="css-1cjt916-Home"
              src="waldo.jpg"
            />
            <img
              class="css-1cjt916-Home"
              src="wizard.jpg"
            />
            <img
              class="css-1cjt916-Home"
              src="wendy.jpg"
            />
            <img
              class="css-1cjt916-Home"
              src="odlaw.jpg"
            />
          </div>
        </div>
      </a>
      <a
        class="css-10dzrpy-Home"
        href="#/puzzles/waldo3"
      >
        <img
          alt="waldo3"
          class="puzzle-img css-e4yxh9-Home"
          loading="lazy"
          src="waldo3.jpg"
        />
        <div
          class="css-1b3wpv8-Home"
        >
          <p>
            Level 
            3
          </p>
          <div
            class="css-bshjkg-Home"
          >
            <img
              class="css-1cjt916-Home"
              src="waldo.jpg"
            />
            <img
              class="css-1cjt916-Home"
              src="wizard.jpg"
            />
            <img
              class="css-1cjt916-Home"
              src="wendy.jpg"
            />
            <img
              class="css-1cjt916-Home"
              src="odlaw.jpg"
            />
          </div>
        </div>
      </a>
      <a
        class="css-10dzrpy-Home"
        href="#/puzzles/waldo4"
      >
        <img
          alt="waldo4"
          class="puzzle-img css-e4yxh9-Home"
          loading="lazy"
          src="waldo4.jpg"
        />
        <div
          class="css-1b3wpv8-Home"
        >
          <p>
            Level 
            4
          </p>
          <div
            class="css-bshjkg-Home"
          >
            <img
              class="css-1cjt916-Home"
              src="waldo.jpg"
            />
            <img
              class="css-1cjt916-Home"
              src="wizard.jpg"
            />
            <img
              class="css-1cjt916-Home"
              src="wendy.jpg"
            />
            <img
              class="css-1cjt916-Home"
              src="odlaw.jpg"
            />
          </div>
        </div>
      </a>
      <a
        class="css-10dzrpy-Home"
        href="#/puzzles/waldo5"
      >
        <img
          alt="waldo5"
          class="puzzle-img css-e4yxh9-Home"
          loading="lazy"
          src="waldo5.jpg"
        />
        <div
          class="css-1b3wpv8-Home"
        >
          <p>
            Level 
            5
          </p>
          <div
            class="css-bshjkg-Home"
          >
            <img
              class="css-1cjt916-Home"
              src="waldo.jpg"
            />
            <img
              class="css-1cjt916-Home"
              src="wizard.jpg"
            />
            <img
              class="css-1cjt916-Home"
              src="wendy.jpg"
            />
            <img
              class="css-1cjt916-Home"
              src="odlaw.jpg"
            />
          </div>
        </div>
      </a>
      <a
        class="css-10dzrpy-Home"
        href="#/puzzles/waldo6"
      >
        <img
          alt="waldo6"
          class="puzzle-img css-e4yxh9-Home"
          loading="lazy"
          src="waldo6.jpg"
        />
        <div
          class="css-1b3wpv8-Home"
        >
          <p>
            Level 
            6
          </p>
          <div
            class="css-bshjkg-Home"
          >
            <img
              class="css-1cjt916-Home"
              src="waldo.jpg"
            />
            <img
              class="css-1cjt916-Home"
              src="wizard.jpg"
            />
            <img
              class="css-1cjt916-Home"
              src="wendy.jpg"
            />
            <img
              class="css-1cjt916-Home"
              src="odlaw.jpg"
            />
          </div>
        </div>
      </a>
    </div>
    <div
      class="Footer"
    >
      <div
        class="made-by css-q4lvbw-Footer"
      >
        <p>
          Made by
        </p>
        <a
          class="css-1ht0kct-Footer"
          href="https://github.com/dwgrossberg"
        >
          <img
            alt="github-icon"
            class="css-18czqf1-Footer"
            id="github-icon"
            src="github.png"
          />
        </a>
        <p>
          Dan Grossberg
        </p>
      </div>
    </div>
  </div>
</div>
`;
